src/enzyme_ad/jax/Passes/EnzymeHLOOpt.cpp

```diff
@@ -6650,6 +6650,61 @@ struct DynamicGatherOpIsNotDynamic
   }
 };
 
+struct ConcatenateGather : public OpRewritePattern<stablehlo::ConcatenateOp> {
+  using OpRewritePattern<stablehlo::ConcatenateOp>::OpRewritePattern;
+
+  LogicalResult matchAndRewrite(stablehlo::ConcatenateOp op,
+                                PatternRewriter &rewriter) const override {
+    if (op.getDimension() != 0)
+      return failure();
```

Comment:
Can this be generalized past dim 0

Reply:
thank! I now use concat const folding and it can be relaxed.